### PR TITLE
Handle zsh glob qualifiers correctly

### PR DESCRIPTION
**Pull request by mrubens**
> [!IMPORTANT]
> Enhance command validation to detect and handle zsh glob qualifiers with code execution, preventing auto-approval of such commands.
> 
>   - **Security**:
>     - Adds detection for zsh glob qualifiers with code execution (e:...:) in `containsDangerousSubstitution()` in `command-validation.ts`.
>     - Updates tests in `command-validation.spec.ts` to cover various zsh glob qualifier patterns and ensure they are flagged as dangerous.
>   - **Validation Logic**:
>     - Modifies `getCommandDecision()` to prevent auto-approval for commands containing zsh glob qualifiers with code execution.
>     - Updates `getSingleCommandDecision()` to handle new dangerous patterns.
>   - **Testing**:
>     - Adds multiple test cases in `command-validation.spec.ts` to verify detection of zsh glob qualifiers and ensure correct decision logic.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=RooCodeInc%2FRoo-Code&utm_source=github&utm_medium=referral)<sup> for d62e2115c9b83e90a7f99fe735c6ebec5eeb3d44. You can [customize](https://app.ellipsis.dev/RooCodeInc/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: webview-ui/src/utils/command-validation.ts [hunk 1]
```diff
@@ -105,13 +106,19 @@ export function containsDangerousSubstitution(source: string): boolean {
 	// =(...) creates a temporary file containing the output of the command, but executes it
 	const zshProcessSubstitution = /=\([^)]+\)/.test(source)
 
+	// Check for zsh glob qualifiers with code execution (e:...:)
+	// Patterns like *(e:whoami:) or ?(e:rm -rf /:) execute commands during glob expansion
+	// This regex matches patterns like *(e:...:), ?(e:...:), +(e:...:), @(e:...:), !(e:...:)
+	const zshGlobQualifier = /[*?+@!]\(e:[^:]+:\)/.test(source)
```
Comment (ghost) [inline webview-ui/src/utils/command-validation.ts:112]: Could we consider if there are other zsh glob patterns that might also support the e: qualifier? For example, what about patterns like **(e:cmd:) for recursive globs? The current regex only matches specific glob operators.

File: webview-ui/src/utils/__tests__/command-validation.spec.ts [hunk 2]
```diff
@@ -292,6 +292,27 @@ ls -la || echo "Failed"`
...
+			// Glob qualifiers in middle of command
+			expect(containsDangerousSubstitution("ls -la *(e:date:) test")).toBe(true)
+
+			// Multiple glob qualifiers
+			expect(containsDangerousSubstitution("cat *(e:whoami:) ?(e:pwd:)")).toBe(true)
+
+			// Glob qualifiers with complex commands
+			expect(containsDangerousSubstitution("ls *(e:open -a Calculator:)")).toBe(true)
+			expect(containsDangerousSubstitution("rm *(e:sudo apt install malware:)")).toBe(true)
+		})
```
Comment (ghost) [inline webview-ui/src/utils/__tests__/command-validation.spec.ts:314]: Consider adding more edge case tests to ensure robustness:
- Empty command within the qualifier: *(e::)
- Nested colons in commands: *(e:echo ":test:":)
- Multiple spaces in commands: *(e:echo    test:)

These edge cases could help ensure the regex handles unusual but valid patterns correctly.

File: webview-ui/src/utils/command-validation.ts [hunk 3]
```diff
@@ -72,6 +72,7 @@ type ShellToken = string | { op: string } | { command: string }
  * - ${!var} - Indirect variable references
  * - <<<$(...) or <<<`...` - Here-strings with command substitution
  * - =(...) - Zsh process substitution that executes commands
+ * - *(e:...:) or similar - Zsh glob qualifiers with code execution
```
Comment (ghost) [inline webview-ui/src/utils/command-validation.ts:75]: Would it be helpful to add a link or reference to zsh documentation about glob qualifiers? This could help developers who want to learn more about this feature and understand the security implications better.

File: webview-ui/src/utils/__tests__/command-validation.spec.ts [hunk 4]
```diff
@@ -292,6 +292,27 @@ ls -la || echo "Failed"`
 			expect(containsDangerousSubstitution("ls =(sudo apt install malware)")).toBe(true)
 		})
 
+		it("detects zsh glob qualifiers with code execution (e:...:)", () => {
```
Comment (ghost) [inline webview-ui/src/utils/__tests__/command-validation.spec.ts:295]: Consider grouping these zsh glob qualifier tests in a dedicated describe block for better test organization, similar to how the zsh process substitution tests are grouped. This would make the test suite more maintainable and easier to navigate.
